pkg/controller/elasticsearch/user/roles.go

```diff
@@ -173,6 +175,22 @@ var (
 				},
 			},
 		},
+		LogstashUserRole: esclient.Role{
+			Cluster: []string{
+				"monitor",
+				"manage_ilm",
+				"read_ilm",
+				"manage_logstash_pipelines",
+				"manage_index_templates",
+				"cluster:admin/ingest/pipeline/get",
+			},
+			Indices: []esclient.IndexRole{
+				{
+					Names:      []string{"logstash-*", "ecs-logstash-*", "logs-*", "metrics-*", "synthetics-*", "traces-*"},
+					Privileges: []string{"manage", "read", "create_doc", "view_index_metadata", "create_index"},
+				},
```

Comment:
I think we need more permissions here, in terms of both index names, and privileges. 
I tried a few different simple scenarios, and there were a few issues - 
Removing `data_streams => false` will write to `logstash`, and `ecs_compatibility => disabled` will write to `ecs-logstash` by default, so we will need to add those to the list of `Names`
Additionally, I think we need to add additional privileges  - we get a 
```
[indices:data/write/index:op_type/index] is unauthorized for user [default-logstash-sample-default-elasticsearch-logstash-user] with effective roles [eck_logstash_user_role] on indices [ecs-logstash], this action is granted by the index privileges [create,index,write,all]"
``` 
error when attempting to write to this index. I suspect we will need the `write` privilege here - the elasticsearch output is able to do `create`, `index`, `update` and `delete` [actions](https://www.elastic.co/guide/en/logstash/current/plugins-outputs-elasticsearch.html#plugins-outputs-elasticsearch-action)
If we are going to limit the indices here, we will need to add instructions on how users can get around this

Reply:
Thanks for double check it. I have updated the setting. 
`ecs-logstash` and `logstash` are required by `ilm_rollover_alias`
The logstash [doc](https://www.elastic.co/guide/en/logstash/master/ls-security.html) regarding secure connection needs update

Reply:
Thank you! This now works as expected.